src/mixins/canvas_grouping.mixin.js

```diff
@@ -13,8 +13,19 @@
      */
     _shouldGroup: function(e, target) {
       var activeObject = this._activeObject;
-      return activeObject && this._isSelectionKeyPressed(e) && target && target.selectable && this.selection &&
-            (activeObject !== target || activeObject.type === 'activeSelection') && !target.onSelect({ e: e });
+      // check if an active object exists on canvas and if the user is pressing the `selectionKey` while canvas supports multi selection.
+      return !!activeObject && this._isSelectionKeyPressed(e) && this.selection
```

Comment:
this.selection should block you from creating multi selections.
It is true that it would be more logic to be constrained to the group selector only, since setting the selectionKey to something that does not exist would allow you to block shift+click and you would keep granularity in choices.
I remember i went over this topic more than once.
<img width="1165" alt="image" src="https://user-images.githubusercontent.com/1194048/162612906-75daf25a-f772-4ca3-b6a3-3dae698797cb.png">

this is old old code, we can eventually change it, we can open an issue or discussion for it.
I'm not attached to it and does not change the complexity in a significant way.

Reply:
So we leave it as is for now, right?

Reply:
yes that change needs to be a PR with that change only because is totally feasible is a beahviour change, has its own test changes, can't be mixed and is completely tangential to this PR